src/NServiceBus.Core/Unicast/MessageHandlerRegistry.cs

```diff
@@ -33,7 +34,7 @@ public IEnumerable<MessageHandler> GetHandlersFor(Type messageType)
             return from handlersAndMessages in handlerAndMessagesHandledByHandlerCache
                 from messagesBeingHandled in handlersAndMessages.Value
                 where Type.GetTypeFromHandle(messagesBeingHandled.MessageType).IsAssignableFrom(messageType)
-                select new MessageHandler(messagesBeingHandled.MethodDelegate, Type.GetTypeFromHandle(handlersAndMessages.Key));
+                select new MessageHandler(messagesBeingHandled.MethodDelegate, Type.GetTypeFromHandle(handlersAndMessages.Key), messagesBeingHandled.MethodInfo);
```

Comment:
This gives us the ability to know the class name / method name of the handler being invoked for friendly messages. Is there another way to do this?